src/main/java/bio/terra/service/snapshot/flight/create/AddSourceDatasetPolicyEmailsIfInheritStewardStep.java

```diff
@@ -0,0 +1,28 @@
+package bio.terra.service.snapshot.flight.create;
+
+import bio.terra.service.auth.iam.IamRole;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.snapshot.flight.SnapshotWorkingMapKeys;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import com.fasterxml.jackson.core.type.TypeReference;
+import java.util.List;
+import java.util.Map;
+
+public interface AddSourceDatasetPolicyEmailsIfInheritStewardStep extends Step {
+  default void addSourceDatasetPolicyEmailsIfInheritSteward(
```

Comment:
Does this need to be an `interface`? If not I think it would be more nature to use `abstract class` here.
```suggestion
public interface AddSourceDatasetPolicyEmailsIfInheritStewardStep extends Step {
  default void addSourceDatasetPolicyEmailsIfInheritSteward(
```

I've used `interface` and `default` with `Step`s before so it can be used as a `record` base type. So if you wanted to make `SnapshotAuthzTabularAclStep` (for example) a `record` then using an `interface` would be necessary here.